### PR TITLE
Fix always loading issue caused by error handling

### PR DESCRIPTION
**Pull request by hwillson**
The React Apollo `Query` component relies on Apollo Client's `ObservableQuery->getCurrentResult` method to get the results of a query to display. When `ObservableQuery` encounters an error, it stores that error in its associated query store, so that it can be passed back to React Apollo for handling. Unfortunately, the error stored in the query store isn't cleared out until a component unmounts, which means components that handle an error response followed by a valid response, won't get the chance to render the valid response. They're stuck rendering the error state, since the error stored in the `ObservableQuery` query store is never removed.

This commit calls into a new `ObservableQuery.resetQueryStoreErrors` method, to clear out previously used errors, before moving on to handle subsequent responses. This means a component can handle an error, then receive and display a valid subsequent response.

Fixes: https://github.com/apollographql/react-apollo/issues/3090

**Note:** This PR is dependent on https://github.com/apollographql/apollo-client/pull/4941 being merged.

### CODE REVIEW

File: src/Query.tsx [hunk 1]
```diff
@@ -349,14 +349,8 @@ export default class Query<TData = any, TVariables = OperationVariables> extends
...
       error: error => {
-        if (!this.lastResult) {
-          // We only want to remove the old subscription, and start a new
-          // subscription, when an error was received and we don't have a
-          // previous result stored. This means either no previous result was
-          // received due to problems fetching data, or the previous result
-          // has been forcefully cleared out.
-          this.resubscribeToQuery();
-        }
+        this.resubscribeToQuery();
```
Comment (hwillson) [inline src/Query.tsx]: I haven't added any additional tests to validate this behavior, as switching to always resubscribe in the error callback here lead to existing test failures. These were then fixed by clearing query store errors below.

**Comment by jasonpaulos** [comment]
@hwillson Any plans to bring this fix to RA 3 as well? Always resubscribing after an error fixes #2070 as well.

**Comment by hwillson** [comment]
@jasonpaulos Definitely! I have it ready (I just haven't committed it yet). I should have it in the RA 3 branch shortly. As soon as it's there, I'm then planning on publishing a final version of RA 2, then will merge the RA 3 branch into `master`. So🤞by tomorrow morning we should be working against only one codebase again (yay! 🙂). Thanks!

**Comment by sambaek** [comment]
any updates on when the final version of RA 2 will be published with this fix?

**Comment by jasonpaulos** [comment]
@sambaek I believe it was just published as 2.5.7

**Comment by developdeez** [comment]
I was using 2.5.6 and it was getting stuck.
I upgraded to 2.5.8 now it blinks several times before showing the page. 
What gives?

**Comment by jasonpaulos** [comment]
Hi @developdeez, could you create a new issue with this bug? It would also help a lot if you could provide a runnable reproduction that shows this. Thanks.

**Pull request merged**